### DISCUSSION TITLE
Sonoff POW320D Power measurement Issues

### DISCUSSION THREAD

**Discussion by zizebra** [arendst/Tasmota#20047] [Q&A]
Hi All
Device Sonoff POW R320D Elite
Program Version	13.2.0(tasmota32)
Build Date & Time	2023-10-19T09:05:13
Core/SDK Version	2_0_14/4.4.6.231011
Uptime	0T02:17:06
Flash Write Count	172
Boot Count	14

I recently got a Sonoff POWR320D which iam using to control my geyser. Prior to connecting the device on geyser i had it calibrated using 60W soldering iron. All seemed fine with power usage values. I connected the device last night on my geyser . Everything else except the power measurement is working as expected. First i noticed this morning that Geyser had used 8Kw but the sonoff device did not register any power usage on todays value and totals.  After searching on the forums i found one case similar to mine where it was suggested to restart the device. I power cycled the device and after the restart the values seems to update but very slowly and the figures seems off. I let it run  for about 1hour with the Geyser running but figure the figures are still very low compared to actual usage actual usage was about 3KW but the device recorded 0.015KW. Iam not sure if i need to calibrate again or change firmware version. 
![image](https://github.com/arendst/Tasmota/assets/4173129/9fca2d9a-fc98-4c61-b597-db623a9bec82)
as can be seen on the image, i noticed that the reactive power is always 0 never changes. How else can i do resolve this.

**Comment by sfromis** [comment]
I'd not recommend a soldering iron as calibration load, as these will often be temperature controlled, and many of them also have a transformer. However, I'm not sure that this would be the only explanation, just something needing to be done right before looking into what may be happening.

**Reply by zizebra** [reply]
Thank you. My soldering iron is old school, plain 60w, no controls and no transformer. Verified previously to be 60w.

**Comment by zizebra** [comment]
I downgraded to ver12.02 which I am using on my other sonoff POWR2 and the values are now updating on reasonable rate.i will monitor and compare with the expected values for the period.

**Comment by arendst** [comment]
Just verified my POWR320D with v3.2.0 and a load of 1700Watts and see no issue.

You might want to check your calibration values which should be around below:
```
14:39:58.308 CMD: voltageset
14:39:58.309 SRC: WebConsole from 192.168.2.171
14:39:58.311 CMD: Grp 0, Cmd 'VOLTAGESET', Idx 1, Len 0, Pld -99, Data ''
14:39:58.318 MQT: stat/powr320d3/RESULT = {"VoltageSetCal":1872}
14:40:03.195 CMD: currentset
14:40:03.196 SRC: WebConsole from 192.168.2.171
14:40:03.198 CMD: Grp 0, Cmd 'CURRENTSET', Idx 1, Len 0, Pld -99, Data ''
14:40:03.210 MQT: stat/powr320d3/RESULT = {"CurrentSetCal":15615}
14:40:07.832 CMD: powerset
14:40:07.834 SRC: WebConsole from 192.168.2.171
14:40:07.836 CMD: Grp 0, Cmd 'POWERSET', Idx 1, Len 0, Pld -99, Data ''
14:40:07.846 MQT: stat/powr320d3/RESULT = {"PowerSetCal":5074}
```

**Reply by zizebra** [reply]
Thank you. Just to validate after having calibrated using my soldering iron. The values were updating  as expected. But when I put on my geyser,  I struggled to get the counter to just 10w. Yet when I downgraded earlier, the device registered 10kw which matches my consumption from the meter. I expected the counter to move and be off by a smaller value. 

Here is my current calibration values

18:46:11.096 CMD: voltageset
18:46:11.098 SRC: WebConsole from 192.168.0.101
18:46:11.101 CMD: Grp 0, Cmd 'VOLTAGESET', Idx 1, Len 0, Pld -99, Data ''
18:46:11.110 MQT: stat/geyser/RESULT = {"VoltageSetCal":1913}
18:46:17.486 CMD: currentset
18:46:17.488 SRC: WebConsole from 192.168.0.101
18:46:17.490 CMD: Grp 0, Cmd 'CURRENTSET', Idx 1, Len 0, Pld -99, Data ''
18:46:17.498 MQT: stat/geyser/RESULT = {"CurrentSetCal":15680}
18:46:24.777 CMD: powerset
18:46:24.779 SRC: WebConsole from 192.168.0.101
18:46:24.781 CMD: Grp 0, Cmd 'POWERSET', Idx 1, Len 0, Pld -99, Data ''
18:46:24.788 MQT: stat/geyser/RESULT = {"PowerSetCal":13083}

**Reply by zizebra** [reply]
I re-applied those same calibration values and seems updating now. It appears that when i used the powerset/currentset/voltageset combination then it does not update the totals. But when i used powercal or voltagecal then values start updating. 

Thank you

**Reply by sfromis** [reply]
When you originally calibrated, did you make sure to do it using current in milliamps, not amps?

**Reply by zizebra** [reply]
I used milliamps, but i can tell you that when i change the values using any of the powerset/currentset/voltageset , at some point, it stops tracking the  totals. However when i use the currentcal/voltagecal/currentcal with exactly the same values derived from running  powerset/currentset/voltageset. Then it starts counting.

**Reply by zizebra** [reply]
just wanted to check what this entry means and if i need to worry or change something 

12:58:10.934 CSE: Checksum failure
12:58:10.971 CSE: Overload
12:58:28.467 CSE: Checksum failure
12:58:29.446 CSE: Overload
12:59:45.069 CSE: Checksum failure
12:59:45.459 CSE: Overload
13:00:18.431 CSE: Overload

**Reply by arendst** [reply]
Checksum failure is an idication of bad comms between the ESP32 and the CSE7766 energy monitoring chip

Overload is shown when the measured load is more than 5.5kW
